Updating the readme.md for the project,

abhinandanjvx opened the pull request.

abhinandanjvx commented on README.md:
````diff
@@ -1 +1,154 @@
+```
+# TCModel
+
+[API Docs](https://www.iabtcf.com/api/core/classes/tcmodel.html)
````
okk i will remove those links. But since we don't have the similar webpages for java library. I will leave them blank as of now. We might need to build the same thing for java.

chrispaterson replied:
Can we generate some Java Docs?

chrispaterson replied:
I can put them up on iabtcf.com

abhinandanjvx commented on README.md:
```diff
@@ -1 +1,154 @@
+
+[API Docs](https://www.iabtcf.com/api/core/classes/tcmodel.html)
+
+### Creating a new TCModel
```
Yes! Understood. I just added them in case someone wants to add the encoder later, So he/she might have an idea about the TCModel class. If required, I will remove them.

chrispaterson replied:
I would remove them.  It's going to cause more problems when people open issues on the repository saying "The documentation says I can do this but it doesn't work."  So the documentation should reflect only what is currently available.

chrispaterson commented on README.md:
````diff
@@ -1 +1,131 @@
+System.out.println(tcModel.getVendorConsents().getSet().size()); // 0
+```
+
+### Setting Publisher Restrictions
````
Let's remove this until we can actually set publisher restrictions, since we don't support encoding

a-s-rose replied:
Hi Chris,

are there plans for the future to encode the TCString with java? Or is the intention to not encode the TCString with java at all?

Thx 
Antje

chrispaterson replied:
Yes the plan is to support encoding in the future, just trying to get over the first hurdle!

abhinandanjvx commented:
Hey @chrispaterson . 

1. I have updated the README.md describing how to import the project through maven central repo
2. Added the support for version 1 decoding and also verified it thoroughly.
3. Refactored the segment encoder classes

chrispaterson commented on README.md:
```diff
@@ -1 +1,55 @@
+<dependency>
+  <groupId>com.iabtcf</groupId>
+  <artifactId>iabtcf-core</artifactId>
+  <version>1.0.4-alpha.0</version>
```
This should be 1.0.2-alpha.0

macfol replied:
Wouldn't it be better to just provide link to maven central? It seems redundant to update the readme every release...

laktech replied:
it's common practice to include a copy-pasteable link to the latest version of the library in the readme. i havn't read through this in-depth, but seems like updating a README.md with a template that injects the version field can be automated,

https://stackoverflow.com/questions/34629326/update-versions-contained-in-readme-on-maven-release

chrispaterson commented on README.md:
````diff
@@ -1 +1,55 @@
+```
+**Note:-** 
+
+ 1. The current library only support decoding of TCString(consent String) to [populate](#using) the TCModel.
````
Change to:
1. The library currently only supports decoding of an encoded TC string (formerly consent string) and [populating](#using) a TCModel.

chrispaterson commented on README.md:
```diff
@@ -1 +1,55 @@
+**Note:-** 
+
+ 1. The current library only support decoding of TCString(consent String) to [populate](#using) the TCModel.
+ 2. The current library supports both GDPR version 1 and version 2 decoding.
```
Does it support v1?  I thought it didn't yet?

chrispaterson replied:
If it does support v1, I would would change the whole note to:
**NOTE** The library currently only supports _**decoding**_ of an encoded v1 Consent string and a v2 TC string and populating a TCModel.

laktech replied:
what is a TCModel?

chrispaterson commented on pom.xml:
```diff
@@ -9,7 +9,7 @@
     <groupId>com.iabtcf</groupId>
     <artifactId>iabtcf-core</artifactId>
-    <version>1.0.3-alpha.0-SNAPSHOT</version>
+    <version>1.0.4-alpha.0-SNAPSHOT</version>
```
1.0.2-alpha.0 
maven central version

chrispaterson commented on src/main/java/com/iabtcf/TCModel.java:
```diff
@@ -503,6 +503,53 @@ public SortedVector getPublisherCustomLegitimateInterest() {
         this.publisherCustomLegitimateInterest = publisherCustomLegitimateInterest;
     }
+    /**
+     * @Description Whenever decoding is failed call this method to reset the tcModel
```
Why would you just not create a new one?

laktech commented:
The README content updates in this PR seem useful but the code has since evolved. If you'd still like to contribute, please review the readme file on the develop branch and re-submit.

laktech commented:
closing this PR due to inactivity
